klee fails with glibc 2.33

openSUSE switched to glibc 2.33 and klee cannot be run with POSIX runtime:
```
******************** TEST 'KLEE :: Runtime/POSIX/Fcntl.c' FAILED ********************
Script:
--
clang-11.0 -Iinclude test/Runtime/POSIX/Fcntl.c -emit-llvm -O0 -Xclang -disable-O0-optnone -c -o build/test/Runtime/POSIX/Output/Fcntl.c.tmp2.bc
rm -rf build/test/Runtime/POSIX/Output/Fcntl.c.tmp.klee-out
build/bin/klee --output-dir=build/test/Runtime/POSIX/Output/Fcntl.c.tmp.klee-out --posix-runtime --exit-on-error build/test/Runtime/POSIX/Output/Fcntl.c.tmp2.bc --sym-files 1 10
--
Exit Code: 1

Command Output (stdout):
--
$ "clang-11.0" "-Iinclude" "test/Runtime/POSIX/Fcntl.c" "-emit-llvm" "-O0" "-Xclang" "-disable-O0-optnone" "-c" "-o" "build/test/Runtime/POSIX/Output/Fcntl.c.tmp2.bc"
# command stderr:
test/Runtime/POSIX/Fcntl.c:11:5: warning: implicit declaration of function 'klee_silent_exit' is invalid in C99 [-Wimplicit-function-declaration]
    klee_silent_exit(0);
    ^
1 warning generated.

$ "rm" "-rf" "build/test/Runtime/POSIX/Output/Fcntl.c.tmp.klee-out"
$ "build/bin/klee" "--output-dir=build/test/Runtime/POSIX/Output/Fcntl.c.tmp.klee-out" "--posix-runtime" "--exit-on-error" "build/test/Runtime/POSIX/Output/Fcntl.c.tmp2.bc" "--sym-files" "1" "10"
# command stderr:
KLEE: NOTE: Using POSIX model: build/runtime/lib/libkleeRuntimePOSIX64_Debug+Asserts.bca
KLEE: output directory is "build/test/Runtime/POSIX/Output/Fcntl.c.tmp.klee-out"
KLEE: Using STP solver backend
error: Linking globals named '__xstat': symbol multiply defined!
```

It looks like `__xstat` was removed from glibc. But that is not the primary problem. Looking into `nm`'s output:
```
$ llvm-nm runtime/lib/libkleeRuntimePOSIX64_Debug+Asserts.bca | grep -w __xstat
---------------- T __xstat
---------------- T __xstat
```

`__xstat` is really multiply defined! It comes from:
```
$ llvm-ar -x runtime/lib/libkleeRuntimePOSIX64_Debug+Asserts.bca
$ for aa in *bc; do llvm-nm $aa|grep -q __xstat && echo $aa; done
fd_3264_Debug+Asserts.bc
fd_6464_Debug+Asserts.bc
```

Both `fd_32.c` and `fd_64.c` are compiled into the 64bit library. That looks like a bug.

--- a/runtime/POSIX/fd_64.c
+++ b/runtime/POSIX/fd_64.c
@@ -73,6 +73,16 @@ int openat(int fd, const char *pathname, int flags, ...) {
   return __fd_openat(fd, pathname, flags, mode);
 }
 
+/* removed in glibc 2.33 */
+#ifdef __REDIRECT_NTH
+extern int __REDIRECT_NTH(__fxstat, (int __ver, int __fildes,
+                          struct stat *__stat_buf), __fxstat64);
+extern int __REDIRECT_NTH(__xstat, (int __ver, const char *__filename,
+                          struct stat *__stat_buf), __xstat64);
+extern int __REDIRECT_NTH(__lxstat, (int __ver, const char *__filename,
+                          struct stat *__stat_buf), __lxstat64);
+#endif
+
 off64_t lseek(int fd, off64_t offset, int whence) {
   return __fd_lseek(fd, offset, whence);
 }

